management-core/src/main/java/com/ns/task/repositories/ProductRepositoryImpl.java

```diff
@@ -21,12 +22,13 @@
     }
 
     @Override
-    public ProductEntity saveProduct(ProductEntity product) {
+    public ProductEntity saveProduct(ProductEntity product)  throws SQLIntegrityConstraintViolationException {
```

Comment:
Why this `SQLIntegrityConstraintViolationException`? How did you know this is the correct one?

Reply:
changed to DataIntegrityViolationException, this one is the abstraction exception that spring throws, I am not sure if this is the correct one